libyul/optimiser/CommonSubexpressionEliminator.cpp

```diff
@@ -89,12 +89,9 @@ void CommonSubexpressionEliminator::visit(Expression& _e)
 		if (m_value.count(name))
 		{
 			assertThrow(m_value.at(name).value, OptimizerException, "");
-			if (holds_alternative<Identifier>(*m_value.at(name).value))
-			{
-				YulString value = std::get<Identifier>(*m_value.at(name).value).name;
-				assertThrow(inScope(value), OptimizerException, "");
-				_e = Identifier{locationOf(_e), value};
-			}
+			if (Identifier const* value = get_if<Identifier>(m_value.at(name).value))
+				if (inScope(value->name))
```

Comment:
The semantics are not the same here, missing the assert.

Reply:
Yeah, that's the point, see https://github.com/ethereum/solidity/pull/10467#discussion_r533790328

Reply:
Ah ok